test_vm/src/util.rs

```diff
@@ -224,23 +230,23 @@ pub fn miner_prove_sector<BS: Blockstore>(
         Some(prove_commit_params),
     );
 
-    ExpectInvocation {
-        to: *miner_id,
-        method: MinerMethod::ProveCommitSector as u64,
-        from: Some(*worker),
-        subinvocs: Some(vec![ExpectInvocation {
-            to: STORAGE_POWER_ACTOR_ADDR,
-            method: PowerMethod::SubmitPoRepForBulkVerify as u64,
-            ..Default::default()
-        }]),
-        ..Default::default()
-    }
-    .matches(v.take_invocations().last().unwrap());
+    // ExpectInvocation {
+    //     to: miner_id,
+    //     method: MinerMethod::ProveCommitSector as u64,
+    //     from: Some(worker),
+    //     subinvocs: Some(vec![ExpectInvocation {
+    //         to: STORAGE_POWER_ACTOR_ADDR,
+    //         method: PowerMethod::SubmitPoRepForBulkVerify as u64,
+    //         ..Default::default()
+    //     }]),
+    //     ..Default::default()
+    // }
+    // .matches(v.take_invocations().last().unwrap());
```

Comment:
Abstraction of invocation expectations seems like it will take a significant amount of effort and generate quite a bit of noise. I'm deferring this for now as I don't have a clear idea of what this should look like. 

Within this util file, I'm not convinced of the value of having these expectations checked - it feels like it's mostly checking the internal consistency of the util functions. Elsewhere, the invocation expectations are mostly checked at the end of a test case. This is convenient for now as it means we can run the expectation on the concrete `TestVM` after the inner test (which should take an abstract `&dyn VM` ) has run. See test_vm/tests/commit_post_test.rs for examples. 

However, where expectation invocations are checked mid-test, this currently forces the tests to be broken up into sections that take a reference to the VM either by &dyn VM (running the bulk of the test) or by TestVM (to check the invocation expectations).

Reply:
Decision was made to bring this into the trait. It will be fairly easy for alternative implementations to comply.